packages/lib-sourcify/src/Compilation/SolidityCompilation.ts

```diff
@@ -202,7 +266,7 @@ export class SolidityCompilation extends AbstractCompilation {
   public async compile(forceEmscripten = false) {
     const contract =
       await this.compileAndReturnCompilationTarget(forceEmscripten);
-    this._metadata = JSON.parse(contract.metadata.trim());
+    this._metadata = JSON.parse(contract.metadata || '{}'.trim());
```

Comment:
We have one place (maybe even more?) in the server where we expect metadata to be present:
https://github.com/ethereum/sourcify/blob/fb6387a9b948f0fa44fd13c2c71a4abc53bc8ea4/services/server/src/server/services/storageServices/SourcifyDatabaseService.ts#L344
We should also adapt this.

Reply:
Besides this, you need braces here:
```suggestion
    this._metadata = JSON.parse((contract.metadata || '{}').trim());
```